requirements.txt

```
@@ -18,3 +18,4 @@ build==0.10
 mypy==1.4.1
 mypy-extensions==1.0
 ruff~=0.6
+pre-commit
```

```suggestion
pre-commit~=3.8
```

This way, we make sure to only use versions compatible with 3.8 (assuming you used the latest available version)